### PR TITLE
Zoom

### PR DESCRIPTION
**Pull request by UriBombastic**
You can now zoom in and move to change your perspective! Borrowed some code from 1000i100 for a problem with ink displacement I otherwise couldn't have solved.

### CODE REVIEW

File: v1.1.1/js/Loopy.js [hunk 1]
```diff
@@ -77,6 +82,8 @@ function Loopy(config){
 		if(!self.modal.isShowing){ // modAl
 			self.model.update(); // modEl
 		}
+		window.addEventListener("wheel", self.zoom);
```
Comment (nlehnert1) [inline v1.1.1/js/Loopy.js:87]: Think we could bind `-` and `=` to be zoom out and zoom in respectively as well? The current implementation works on a laptop trackpad, but it's kinda unwieldy. The ability to take predetermined step size zooms would be helpful.

**Review by nlehnert1 — CHANGES_REQUESTED** [review]
This all seems to work *relatively* well, but it certainly feels a little buggy sometimes. Here's some observations I made that I think should be addressed:
- It would be helpful to have a zoom slider somewhere, where the user can click and drag the slider to adjust zoom level
- A "reset to default zoom" button would also be handy
- It seems like for some reason, within a second or two of the mouse ceasing to move, the page doesn't react to the zoom or pan stuff? It does seem to queue up actions, as they'll all suddenly happen once the mouse is moved, but there is a definite phenomena where you can pan/zoom all you want for ~2 sec after moving the mouse, then it stops drawing on new button presses. I think this is tied to the "draw countdown" timer or whatever it is, I believe located in Model.js

If you could look into these and either address them in this PR or let me know if they're complex enough to warrant their own issue, I'd appreciate it!

File: v1.1.1/js/Loopy.js (hunk 1 above)
Comment (UriBombastic) [reply v1.1.1/js/Loopy.js:87]: Good idea I'll try to implement that

Comment (nlehnert1) [reply v1.1.1/js/Loopy.js:87]: Coolio, thanks!

**Comment by bbuckwalter2** [comment]
I resolved the redraw issue similar to how Node.js did it originally (publish a "mousemove" event).

There are a few other ideas that came up while working on this:

- Make middle click or right click allow for drag panning as that matches my expectations from other software.
- Instead of adding separate event listeners, use the "key/" subscribe and bindings from Key.js to match the rest of the program.

File: v1.1.1/js/Loopy.js [hunk 2]
```diff
@@ -77,6 +82,9 @@ function Loopy(config){
...
+		window.addEventListener("keydown", self.buttonZoom);
+		window.addEventListener("keydown", self.moveWindow);
```
Comment (bbuckwalter2) [inline v1.1.1/js/Loopy.js:88]: Currently, these event listener registrations are happening within an update function. While this doesn't end up registering the same event more than once, it is probably better to have a single registration somewhere that runs once.

Additionally, there is the possibility of using the "subscribe" method from minpubsub to match other code within the project.

**Review by nlehnert1 — APPROVED** [review]
Looks good to me! I went ahead and added some code to prevent the default browser save and print functions, since there's already a custom save and I noticed that the print just looks bad. I'll go ahead and merge it in!

**Pull request merged**